linalg/solvers.cpp

```diff
@@ -232,7 +232,7 @@ void OperatorJacobiSmoother::Mult(const Vector &x, Vector &y) const
    MFEM_FORALL(i, height, Y[i] += DI[i] * R[i]; );
 }
 
-OperatorChebyshevSmoother::OperatorChebyshevSmoother(Operator* oper_,
+OperatorChebyshevSmoother::OperatorChebyshevSmoother(const Operator* oper_,
```

Comment:
Is there a compelling reason to keep this argument a pointer? The other similar methods use references and internally there is an abort if `oper` is NULL.

Reply:
No, I think you're right that the correct interface here would be to pass `oper` by reference since it should never be NULL. However, I think we're stuck with the current version for backwards compatibility (we could add alternatives that accept `const Operator &` but I'm not sure it would be worth the effort).

Perhaps we can keep track of this and other things that would be nice to change but break backwards compatibility so that we could make this change at some point (e.g. next major version release).

Reply:
Little things like this are so easily forgotten. @tzanio, do we have something like a todo list for the next release?

I'd be happy to put together a small PR to deprecate these constructors and add new ones with const references.

Reply:
Good point @mlstowell. We don't have a specific list yet... how about starting an mfem-4.4 todo list [discussion](https://github.com/mfem/mfem/discussions)?

Reply:
see https://github.com/mfem/mfem/discussions/2365